lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -2261,5 +2261,50 @@ LogicalResult TruncIOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// FPToSIOp
+//===----------------------------------------------------------------------===//
+LogicalResult FPToSIOp::verify() {
+  ChannelType srcType = getIn().getType();
+  ChannelType dstType = getOut().getType();
+
+  if (srcType.getDataBitWidth() < dstType.getDataBitWidth()) {
+    return emitError() << "result channel's data type " << dstType.getDataType()
+                       << " must be narrower than operand type "
+                       << srcType.getDataType();
+  }
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// FPToSIOp
+//===----------------------------------------------------------------------===//
+LogicalResult SIToFPOp::verify() {
+  ChannelType srcType = getIn().getType();
+  ChannelType dstType = getOut().getType();
+
+  if (srcType.getDataBitWidth() < dstType.getDataBitWidth()) {
+    return emitError() << "result channel's data type " << dstType.getDataType()
+                       << " must be narrower than operand type "
+                       << srcType.getDataType();
+  }
+  return success();
+}
```

Comment:
I don't think these two verification functions are relevant in this case. The `arith` version of these operations ([here](https://mlir.llvm.org/docs/Dialects/ArithOps/#arithsitofp-arithsitofpop) and [here](https://mlir.llvm.org/docs/Dialects/ArithOps/#arithfptosi-arithfptosiop)) seem to have no constraints on the relation between the respective bitwidth of the argument and result.